File: Small_ProgrammingLang/SourceCode/SAL_VM/MachineUI/VM_MachineUI.vb
Imports SDK.SAL

Public Class VM_MachineUI

    Public Function ExecuteCode(ByRef Code As String) As String
        Return X86API.RunMachineCode(Code)
    End Function

    Private Sub ButtonRunCode_Click(sender As Object, e As EventArgs) Handles ButtonRunCode.Click
        Dim CodeAnalysis As String = RichTextBoxCodeEntry.Text
        RichTextBoxInfo.Text &= ExecuteCode(CodeAnalysis)
    End Sub

    Private Sub ButtonNewScrn_Click(sender As Object, e As EventArgs) Handles ButtonNewScrn.Click
        Dim frm As New VM_MachineUI
        frm.Show()
    End Sub

    Private Sub ButtonClrScrn_Click(sender As Object, e As EventArgs) Handles ButtonClrScrn.Click
        RichTextBoxInfo.Clear()
    End Sub
    Public Sub DISPLAY_TEXT(ByRef Str As String)
        RichTextBoxInfo.Clear()
        RichTextBoxInfo.Text = Str
    End Sub
    Private Sub ButtonRef_Click(sender As Object, e As EventArgs) Handles ButtonRef.Click
        Dim frm As New VM_MachineUI
        frm.DISPLAY_TEXT(My.Resources.QuickRef_SAL)
        frm.Show()
    End Sub

    Private Sub ButtonEMU_Click(sender As Object, e As EventArgs) Handles ButtonEMU.Click
        Dim frmU As New EMU_MachineUI
        frmU.Show()
    End Sub
End Class

File: Small_ProgrammingLang/C_/SourceCode/MachineUI/VM_MachineUI.cs
using System;
using SDK.SAL;

namespace SDK
{
    public partial class VM_MachineUI
    {
        public VM_MachineUI()
        {
            InitializeComponent();
            _ButtonRunCode.Name = "ButtonRunCode";
            _ButtonClrScrn.Name = "ButtonClrScrn";
            _ButtonNewScrn.Name = "ButtonNewScrn";
            _ButtonRef.Name = "ButtonRef";
            _ButtonEMU.Name = "ButtonEMU";
        }

        public string ExecuteCode(ref string Code)
        {
            return X86API.RunMachineCode(ref Code);
        }

        private void ButtonRunCode_Click(object sender, EventArgs e)
        {
            string CodeAnalysis = RichTextBoxCodeEntry.Text;
            RichTextBoxInfo.Text += ExecuteCode(ref CodeAnalysis);
        }

        private void ButtonNewScrn_Click(object sender, EventArgs e)
        {
            var frm = new VM_MachineUI();
            frm.Show();
        }

        private void ButtonClrScrn_Click(object sender, EventArgs e)
        {
            RichTextBoxInfo.Clear();
        }

        public void DISPLAY_TEXT(ref string Str)
        {
            RichTextBoxInfo.Clear();
            RichTextBoxInfo.Text = Str;
        }

        private void ButtonRef_Click(object sender, EventArgs e)
        {
            var frm = new VM_MachineUI();
            string argStr = My.Resources.Resources.QuickRef_SAL;
            frm.DISPLAY_TEXT(ref argStr);
            My.Resources.Resources.QuickRef_SAL = argStr;
            frm.Show();
        }

        private void ButtonEMU_Click(object sender, EventArgs e)
        {
            var frmU = new EMU_MachineUI();
            frmU.Show();
        }
    }
}